### PR TITLE
Introduce ColumnSelector instead of ColumnDescriptor

### PR DESCRIPTION
**Pull request by jleibs**
### What
Resolves:
- https://github.com/rerun-io/rerun/issues/7365
- https://github.com/rerun-io/rerun/issues/7435

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7444?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7444?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7444)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/store/re_chunk_store/src/dataframe.rs [hunk 1]
```diff
@@ -398,6 +398,153 @@ impl ComponentColumnDescriptor {
...
+            component: desc.component_name,
+        }
+    }
+}
+
+/// Select a time column.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TimeColumnSelector {
+    /// The name of the timeline.
+    pub timeline: TimelineName,
```
Comment (abey79) [inline crates/store/re_chunk_store/src/dataframe.rs:462]: I like that these `*Selector` have the simplest possible types. Having extact 1:1 fbs counterparts will make everyone's life easy 👍🏻

File: crates/store/re_chunk_store/src/dataframe.rs [hunk 2]
```diff
@@ -398,6 +398,153 @@ impl ComponentColumnDescriptor {
...
+    pub component: ComponentName,
+}
+
+impl From<ControlColumnDescriptor> for ControlColumnSelector {
+    fn from(desc: ControlColumnDescriptor) -> Self {
+        Self {
+            component: desc.component_name,
+        }
+    }
+}
```
Comment (abey79) [inline crates/store/re_chunk_store/src/dataframe.rs:456]: A helper for the row id column would be neat here.

File: crates/store/re_chunk_store/src/dataframe.rs [hunk 3]
```diff
@@ -398,6 +398,153 @@ impl ComponentColumnDescriptor {
...
+
+impl ComponentColumnSelector {
+    pub fn new<C: re_types_core::Component>(entity_path: EntityPath) -> Self {
+        Self {
+            entity_path,
+            component: C::name(),
+            join_encoding: JoinEncoding::default(),
+        }
+    }
```
Comment (abey79) [inline crates/store/re_chunk_store/src/dataframe.rs:508]: Nit: from my perspective, this is a bit too restrictive an API to deserve the privileged `new()` name. Oftentimes, I just have a component name (e.g. when deserialising from blueprint).

```suggestion
    pub fn new(entity_path: EntityPath, component: ComponentName) -> Self {
        Self {
            entity_path,
            component,
            join_encoding: JoinEncoding::default(),
        }
    }

    pub fn new_for_component<C: re_types_core::Component>(entity_path: EntityPath) -> Self {
        Self::new(entity_path, C::name())
    }
```

File: crates/store/re_chunk_store/src/dataframe.rs [hunk 4]
```diff
@@ -622,6 +769,98 @@ impl ChunkStore {
...
+            });
+
+        // TODO(#6889): Fill `archetype_name`/`archetype_field_name` (or whatever their
+        // final name ends up being) once we generate tags.
+        ComponentColumnDescriptor {
+            entity_path: selector.entity_path.clone(),
+            archetype_name: None,
+            archetype_field_name: None,
+            component_name: selector.component,
+            store_datatype: ArrowListArray::<i32>::default_datatype(datatype.clone()),
```
Comment (abey79) [inline crates/store/re_chunk_store/src/dataframe.rs:833]: Shouldn't that depend on `selector.join_encoding`?

**Review by abey79 — APPROVED** [review]
I like this new API, it fits the dataframe view well. And I can  see a path to have the future `ArchetypeSelector` usable there as well.

File: crates/store/re_chunk_store/src/dataframe.rs (hunk 2 above)
Comment (jleibs) [reply crates/store/re_chunk_store/src/dataframe.rs:456]: Ha, yeah, I did exactly that in the python APIs.

File: crates/store/re_chunk_store/src/dataframe.rs (hunk 3 above)
Comment (jleibs) [reply crates/store/re_chunk_store/src/dataframe.rs:526]: I think you are practically speaking in the minority of users that would feel restricted here.  I think having concrete components is likely to be the more common pattern than dealing with raw ComponentNames.

File: crates/store/re_chunk_store/src/dataframe.rs (hunk 4 above)
Comment (jleibs) [reply crates/store/re_chunk_store/src/dataframe.rs:861]: Nope. The `store_datatype` is always a list. `join_encoding` influences the construction of `.returned_datatype()`

File: crates/store/re_chunk_store/src/dataframe.rs (hunk 3 above)
Comment (abey79) [reply crates/store/re_chunk_store/src/dataframe.rs:526]: Fair enough.

**Pull request merged**